core/context.go

```diff
@@ -186,6 +227,11 @@ func (c *Context) UpdateGroupID(name string, id string) error {
 	return nil
 }
 
+// UserInGroup returns whether a user is in a particular group
+func (c *Context) UserInGroup(user User, group string) bool {
+	return contains(user.Groups, group)
```

Comment:
FYI: we're targeting Go 1.16 according to go.mod, but if we were targeting 1.21, there would be a convenient `slices` package.

Reply:
> FYI: we're targeting Go 1.16 according to go.mod

This can safely be bumped to 1.22

Reply:
I think that should be in a separate PR; seems out of scope for this one.